File: packages/ui-flutter/lib/src/modal.dart
import 'package:flutter/widgets.dart';

import 'dialog.dart';

/// Mirrors `packages/ui/src/components/modal.tsx`, a Dialog recipe with a
/// required title. Built from KinetixDialog + the KinetixDialog* header
/// parts (same call as the other ports).
class KinetixModal extends StatelessWidget {
  const KinetixModal({
    super.key,
    required this.visible,
    required this.onDismiss,
    required this.title,
    required this.child,
  });

  final bool visible;
  final VoidCallback onDismiss;
  final String title;
  final Widget child;

  @override
  Widget build(BuildContext context) {
    return KinetixDialog(
      visible: visible,
      onDismiss: onDismiss,
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        mainAxisSize: MainAxisSize.min,
        children: [
          KinetixDialogHeader(children: [KinetixDialogTitle(title)]),
          const SizedBox(height: 16),
          child,
        ],
      ),
    );
  }
}

File: packages/ui-swiftui/Sources/KinetixUI/Modal.swift
//
// Modal.swift — KinetixModal.
//
// Mirrors packages/ui/src/components/modal.tsx, a Dialog recipe with a
// required title and a corner close. Built from KinetixDialog + the
// KinetixDialog* header parts (same call as the Compose port's
// `KinetixModal` from `KinetixDialog` + `KinetixButton`).
//

import SwiftUI

public struct KinetixModal<Content: View>: View {
    @Binding private var isPresented: Bool
    private let title: String
    private let content: Content

    public init(isPresented: Binding<Bool>, title: String, @ViewBuilder content: () -> Content) {
        self._isPresented = isPresented
        self.title = title
        self.content = content()
    }

    public var body: some View {
        KinetixDialog(isPresented: $isPresented) {
            KinetixDialogHeader {
                KinetixDialogTitle(title)
            }
            content
        }
    }
}
